src/opm/input/eclipse/EclipseState/Tables/SimpleTable.cpp

```diff
@@ -82,37 +84,14 @@ namespace Opm {
         return col[row];
     }
 
-    void SimpleTable::init( const DeckItem& deckItem ) {
-        this->addColumns();
-
-        if ( (deckItem.data_size() % numColumns()) != 0)
-            throw std::runtime_error("Number of columns in the data file is"
-                    "inconsistent with the ones specified");
-
-        size_t rows = deckItem.data_size() / numColumns();
-        for (size_t colIdx = 0; colIdx < numColumns(); ++colIdx) {
-            auto& column = getColumn( colIdx );
-            for (size_t rowIdx = 0; rowIdx < rows; rowIdx++) {
-                size_t deckItemIdx = rowIdx*numColumns() + colIdx;
-                if (deckItem.defaultApplied(deckItemIdx))
-                    column.addDefault( );
-                else if (m_jfunc) {
-                    column.addValue( deckItem.getData<double>()[deckItemIdx] );
-                }
-                else
-                    column.addValue( deckItem.getSIDouble(deckItemIdx) );
-            }
-            if (colIdx > 0)
-                column.applyDefaults(getColumn( 0 ));
-        }
-    }
-
     void SimpleTable::init( const DeckItem& deckItem, double scaling_factor) {
         this->addColumns();
 
         if ( (deckItem.data_size() % numColumns()) != 0)
-            throw std::runtime_error("Number of columns in the data file is"
-                    "inconsistent with the ones specified");
+            throw std::runtime_error(fmt::format("Number of columns in the data file {} is"
+                                                 " inconsistent with the specified item size {}",
+                                                 deckItem.data_size(),
+                                                 numColumns()));
```

Comment:
This message is a clear improvement, but I think we could probably improve it further by splitting `data_size()` into (one-based) row and column indices.  For instance something like
```C++
const auto row = deckItem.data_size() / this->numColumns();
const auto ncol = deckItem.data_size() % this->numColumns();

throw std::runtime_error {
    fmt::format("Number of column items ({}) in row {} of table input "
                "does not match table's specified number of columns ({})"
                ncol + 1, row + 1, this->numColumns())
};
```
Of course the (row,column) index pair will be wrong if the real cause is that some earlier row has too many (or too few) items, but I believe it will nevertheless direct the user's attention to the general area.

Ideally I think we could have the table ID here too, but that will probably require changing the API into something like
```C++
void SimpleTable::init(const DeckItem& deckItem, const double scaling_factor, const int tableID)
```
since the `DeckItem` does not currently record such information.

Reply:
Oh now I noticed you introduced a bug, it should not be ncol + 1 should it.

Reply:
Also come to think of it I think this is more confusing than anything. The error message assumes the error is in the final row. While it's the only thing we can assume there is no guarantee that reflects the actual issue.

Reply:
That is overly verbose and implies we actually handle the table. How about a simple
```throw std::runtime_error {
    fmt::format("Number of input table elements ({}) is "
                "not a multiple of table's specified number "
                "of columns ({})",
                deckItem.data_size(), this->numColumns())
};

Reply:
Okay, I will do it but separately.

Reply:
> I will [add table IDs] separately.

Cool, thanks a lot.  And I've revised my opinion on the API from before.  The table ID should come before the scaling factor,
```C++
void SimpleTable::init(const DeckItem& deckItem, const int tableID, const double scaling_factor)
```
since we always need the table ID, but could use a defaulted scaling factor.

Reply:
> How about a simple
>
> ```C++
> throw std::runtime_error {
>     fmt::format("Number of input table elements ({}) is "
>                 "not a multiple of table's specified number "
>                 "of columns ({})",
>                 deckItem.data_size(), this->numColumns())
> };
> ```

Okay, that's a decent compromise.